### ISSUE TITLE
Material-sidenav tab order and focus needs to be handled

### ISSUE THREAD

**Issue by marcysutton** [angular/material#330]
To be accessible, `material-sidenav` needs to hook into transition events to hide and show the component with `display: none` after animating. This will remove interactive controls from the tab order when they are not in use, which is easier and more performant than changing the `tabIndex` of every interactive child element. 

In addition, focus should be sent into the sidenav when it opens, directing the user's attention to new content on screen. This helps a desktop keyboard user with a smaller screen, as well as a mobile screen reader user. The pattern is very similar to `material-dialog`.
